src/MoBi.Presentation/Presenter/HierarchicalStructureEditPresenter.cs

```diff
@@ -34,14 +34,12 @@ public virtual void Select(ObjectBaseDTO objectBaseDTO)
             RaiseUserDefinedSelectedEvent();
 
          else
-            raiseEntitySelectedEvent(objectBaseDTO);
+            RaiseEntitySelectedEvent(objectBaseDTO);
       }
 
-      private void raiseEntitySelectedEvent(ObjectBaseDTO objectBaseDTO)
+      protected virtual void RaiseEntitySelectedEvent(ObjectBaseDTO objectBaseDTO)
       {
-         // First and second neighbor node selections should not trigger an EntitySelectedEvent
-         // because they are not a selectable entity
-         if(objectBaseDTO.ObjectBase != null)
+         if(!(objectBaseDTO is NeighborDTO))
```

Comment:
This is funky. IHere you know that the DERIVED Class is going to overwrite for NEighbor. .. this is why you are bypassing
It's a bit of leak but why not. However what about a comment explaining why this is excluded like this?